### PR TITLE
Chore/z searchbar

### PR DESCRIPTION
**Pull request by vpasquino**
# Feature - ZSearchbar

## 

### Priority

- [ ] 1 - Highest
- [X] 2 - High
- [ ] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Component (add a Component as approved by Design System)
- [ ] Docs (add documentation)
- [ ] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### Abstract

https://www.figma.com/file/w51ZwDwLNEt0yamar2sRRQ/branch/H83cCjre8v6yis2RZY6toC/Nuovi-Componenti-e-Pattern?node-id=298%3A2&t=oPx9alNVgdHG8BEi-0 

## 

## Component and Fix approval flow to Master branch

A Pull Request to be merged must have two approvals, one from a member of the product team who developed it and another from a member of the dst dev team other than the team representative.

### CODE REVIEW

File: src/components/inputs/z-searchbar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+    }
+
+    return groupedItems;
+  }
+
+  private checkResultsCount(counter: number): boolean {
+    if (!this.currResultsCount) {
+      return true;
+    }
+    if (counter < this.currResultsCount) {
```
Comment (francesco-santi-1) [inline src/components/inputs/z-searchbar/index.tsx:174]: Le condizioni di riga 171 e 174 si possono unire con un `or`?

File: src/components/inputs/z-searchbar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+            class="item-label"
+            title={item.label}
+            innerHTML={this.renderItemLabel(item.label)}
+          />
+        </span>
+      </z-list-element>
+    );
+  }
+
+  private renderItemLabel(label: string): string {
```
Comment (francesco-santi-1) [inline src/components/inputs/z-searchbar/index.tsx:338]: Si potrebbe forse usare questo stesso naming, togliendo quindi "results", anche per `renderResultsItem`, `renderResultsItems` e `renderResultsItemCategory`? Tanto mi pare non ci siano altri item in giro.

File: src/components/inputs/z-searchbar/styles.css [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+:host > div > div.results .item-show-all {
+  display: block;
+  text-align: center;
+}
+
+/* Tablet breakpoint */
+@media (min-width: 768px) {
+  :host > div > div.results .category-heading {
+    font-size: 16px;
```
Comment (francesco-santi-1) [inline src/components/inputs/z-searchbar/styles.css:151]: `--font-size-3` qui e a riga 159? Poi eventualmente `--font-size-2` alle righe 72 e 111.

**Review by BigOoga — APPROVED** [review]

**Review by francesco-santi-1 — APPROVED** [review]

File: src/components/inputs/z-searchbar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+      return;
+    }
+
+    this.emitSearchSubmit();
+  }
+
+  private handleItemClick(item: SearchbarItem): void {
+    this.emitSearchItemClick(item);
+    if (item?.link) {
+      window.location.href = item.link;
```
Comment (saldomik) [inline src/components/inputs/z-searchbar/index.tsx:202]: io questo comportamento lo farei decidere solo all'applicazione che usa la searchbar, emetterei solo l'evento

**Review by saldomik — APPROVED** [review]

**Pull request merged**